src/lib.rs

```diff
@@ -4125,7 +4146,18 @@ pub unsafe trait IntoBytes {
             // than panicking.
             return Err(SizeError::new(self));
         };
-        bytes.copy_from_slice(self.as_bytes());
+        // SAFETY: Through fallible subslicing of `bytes`, we have ensured that
+        // `bytes.len()` is equal to `mem::size_of_val(self)`.
+        //
+        // Neither the size of the source nor the size of the destination change
+        // between the above subslicing operation and the invocation of
+        // `copy_unchecked`. The size of `self` (the source) cannot change,
+        // because `Self` is bounded as `Immutable`, and the size of `bytes`
+        // cannot change, because we hold an exclusive reference to it and do
+        // not change its size.
+        unsafe {
+            util::copy_unchecked(self.as_bytes(), bytes);
```

Comment:
Same here as above - do `self.as_bytes()` first and bounds check using `self_bytes.len()`.